addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf

```diff
@@ -32,20 +34,29 @@ private _duty = GVAR(animDuty);
     };
 } forEach (GVAR(dutyList) select 1);
 
-if (GVAR(isSwimming)) then {
-    _terrainGradient = 0;
+if (!GVAR(isSwimming)) then {
+    private _movementVector = vectorNormalized (velocity _unit);
+    private _side = vectorNormalized(_surfaceNormal vectorCrossProduct [0, 0, 1]);
+    private _up = vectorNormalized(_surfaceNormal vectorCrossProduct _side);
```

Comment:
Would love if we had debug lines just like the grenade throwing for this.

Reply:
I have them locally. Let me add them.